rust/tw_evm/src/abi/encode.rs

```diff
@@ -4,15 +4,10 @@
 // terms governing use, modification, and redistribution, is contained in the
 // file LICENSE at the root of the source code distribution tree.
 
-#include "ParamBase.h"
+use crate::abi::token::Token;
+use tw_memory::Data;
 
-namespace TW::Ethereum::ABI {
-
-// Default implementation for simple types: return encoded value (32 bytes)
-Data ParamBase::hashStruct() const {
-    Data encoded;
-    encode(encoded);
-    return encoded;
+pub fn encode_tokens<I: IntoIterator<Item = Token>>(tokens: I) -> Data {
+    let ethabi_tokens: Vec<_> = tokens.into_iter().map(Token::into_ethabi_token).collect();
+    ethabi::encode(&ethabi_tokens)
 }
```

Comment:
this diff is strange

Reply:
Oh right, don't see the reason for that
Probably, IDE did this for a mistake